Assignment: Project-2- Hypothesis Testing


---


Objective:
To dive deep into the  data to find some valuable insights.


---


Domain: Healthcare

# **Exploratory Data analysis (EDA)**

1. Import the necessary libraries

In [0]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import math
import seaborn as sns
import scipy.stats as stats
import statsmodels.api as sm
from sklearn.preprocessing import LabelEncoder
%matplotlib inline

2. Read the data as a data frame

In [11]:
Data  = pd.read_csv('/insurance.csv') # Import the dataset named 'insurance.csv'
Data.head()  # view the first 5 rows of the data


age     sex     bmi  children smoker     region      charges
0   19  female  27.900         0    yes  southwest  16884.92400
1   18    male  33.770         1     no  southeast   1725.55230
2   28    male  33.000         3     no  southeast   4449.46200
3   33    male  22.705         0     no  northwest  21984.47061
4   32    male  28.880         0     no  northwest   3866.85520

3. Perform basic EDA which should include the following
and print out your insights at every step.



> a. Shape of the data



In [25]:
rows, cols = Data.shape # see the shape of the data
print (f"There are {rows} observations / rows and {cols} attributes / columns")

1338
There are 1338 observations / rows and 7 attributes / columns


> b. Data type of each attribute

In [13]:
Data.info() # To see the data type of each of the variable, number of values entered in each of the variables

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1338 entries, 0 to 1337
Data columns (total 7 columns):
age         1338 non-null int64
sex         1338 non-null object
bmi         1338 non-null float64
children    1338 non-null int64
smoker      1338 non-null object
region      1338 non-null object
charges     1338 non-null float64
dtypes: float64(2), int64(2), object(3)
memory usage: 73.3+ KB


The data has 1338 instances with 7 attributes / columns. 2 integer type, 2 float type and 3 object type (Strings)

> c. Checking the presence of missing values

In [31]:
Data.isna().sum()  # get the count of na values

age         0
sex         0
bmi         0
children    0
smoker      0
region      0
charges     0
dtype: int64

No null values in the dataset

In [0]:
Data